mithril-fake-aggregator/src/main.rs

```diff
@@ -0,0 +1,236 @@
+    Internal(anyhow::Error),
+
+    /// Resource not found (specify what)
+    NotFound(String),
+}
+
+/// Tell axum how to convert `AppError` into a response.
+impl IntoResponse for AppError {
+    fn into_response(self) -> Response<Body> {
+        match self {
+            Self::Internal(e) => {
+                error!("{}", e);
+
+                (StatusCode::INTERNAL_SERVER_ERROR, format!("Error: {:?}", e)).into_response()
+            }
+            Self::NotFound(resource) => (
+                StatusCode::NOT_FOUND,
+                format!("resource '{resource}' not found"),
+            )
+                .into_response(),
+        }
+    }
+}
+
+/// This enables using `?` on functions that return `Result<_, anyhow::Error>` to turn them into
+/// `Result<_, AppError>`. That way you don't need to do that manually.
+impl<E> From<E> for AppError
+where
+    E: Into<anyhow::Error>,
+{
+    fn from(err: E) -> Self {
+        Self::Internal(err.into())
+    }
+}
+
+pub struct OsSignalHandler;
+
+impl OsSignalHandler {
+    pub async fn handle_signal(mut signals: Signals) {
+        while let Some(signal) = signals.next().await {
+            match signal {
+                SIGTERM | SIGINT | SIGQUIT => {
+                    warn!("Signal caught: {signal}, exiting.");
+
+                    break;
+                }
+                SIGHUP => {
+                    info!("SIGHUP signal caught.");
+                }
+                _ => unreachable!(),
+            }
+        }
+    }
+}
+
+#[tokio::main]
+async fn main() -> StdResult<()> {
+    let params = CliArguments::parse();
+    tracing_subscriber::fmt()
+        .with_max_level(params.get_verbosity_level())
+        .init();
+
+    let result = Application::run(params).await;
+
+    match &result {
+        Err(e) => error!("{e}"),
+        Ok(_) => debug!("soft terminated"),
+    };
+
+    result
+}
+struct Application;
+
+impl Application {
+    pub async fn run(params: CliArguments) -> StdResult<()> {
+        info!(
+            "starting Fake Aggregator version {}",
+            env!("CARGO_PKG_VERSION")
+        );
+
+        trace!("setting up signal hook…");
+        // supported signals
+        let signals = Signals::new([SIGTERM, SIGINT, SIGQUIT, SIGHUP])?;
+
+        // launch signal detector
+        let signal_handler = signals.handle();
+
+        trace!("setting up shared state…");
+        let shared_state: SharedState = match params.data_directory {
+            Some(directory) => {
+                info!("Read data files from directory '{}'.", directory.display());
+                AppState::from_directory(&directory)?.into()
+            }
+            None => {
+                debug!("Using default data set.");
+                AppState::default().into()
+            }
+        };
+
+        trace!("configuring router…");
+        let app = Router::new()
+            .route("/aggregator/epoch-settings", get(handlers::epoch_settings))
```

Comment:
Is the "/aggregator/" prefix necessary?

Reply:
No, it could be bound by the configuration, I will change that 👍🏻